src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -574,42 +649,12 @@ namespace OpenMS
             tmp.setMZ(mz);
             spectrum.push_back(tmp);
 
-            //add meta data
-            UInt meta_float_array_index = 0;
-            UInt meta_int_array_index = 0;
-            UInt meta_string_array_index = 0;
-            for (Size i = 0; i < input_data.size(); i++) //loop over all binary data arrays
+            // Only if there are more than 2 data arrays, we need to check
```

Comment:
ok. I was just wondering if there might be some extensions like ion mobility or similar